src/views/QuestionnareModal/QuestionnareModal.tsx

```diff
@@ -253,10 +289,24 @@ export default function QuestionnareModal({
         })
         .catch((error) => {
           console.error('Error fetching data:', error)
+          navigate(Routes.SIGNIN, {
+            replace: true,
+            state: {
+              message: 'Your session has expired. Please log in again.',
+            },
+          })
         })
       axiosInstance
         .get(`scores?datacallid=2&fismasystemid=${system.fismasystemid}`)
         .then((res) => {
+          if (res.status !== 200 && res.status.toString()[0] === '4') {
+            navigate(Routes.SIGNIN, {
+              replace: true,
+              state: {
+                message: 'Your session has expired. Please log in again.',
+              },
```

Comment:
code is very repetitive. As it is, if the message needed to be changed, you would have to do it in many places. I would move all of these state message objects to a single function that gets called. It would accept a parameter to indicate whether to prepend "your changes were not saved"